extensions/stats/plugin_test.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/stats/plugin.h"
+#include "absl/hash/hash_testing.h"
+#include "gtest/gtest.h"
+
+// WASM_PROLOG
+#ifndef NULL_PLUGIN
+#include "api/wasm/cpp/proxy_wasm_intrinsics.h"
+
+#else  // NULL_PLUGIN
+
+#include "extensions/common/wasm/null/null.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+namespace Null {
+namespace Plugin {
+#endif  // NULL_PLUGIN
+
+// END WASM_PROLOG
+
+namespace Stats {
+
+TEST(IstioDimensions, Hash) {
+  IstioDimensions d1;
+  IstioDimensions d2{.request_protocol = "grpc"};
+  IstioDimensions d3{.request_protocol = "grpc", .response_code = "200"};
```

Comment:
this is strange syntax. is this referring to field with a prefix or is the dot a part of the field name?

Reply:
This is a literal struct. Is there a better way?

Reply:
Ok, got it. Looks fine.

Reply:
This syntax is causing warnings on Gcc 7.3x:

"extensions/stats/plugin_test.cc:44:48: sorry, unimplemented: non-trivial designated initializers not supported
   IstioDimensions d2{.request_protocol = "grpc"};"

Is it possible to do something as follows:

/=============================/
  IstioDimensions d1,d2,d3,d4,d5,d6,d7,d8;
  d2.request_protocol = "grpc";
  d3.request_protocol = "grpc";
  d3.response_code = "200";
  d4.request_protocol = "grpc";
  d4.response_code = "400";
  d5.request_protocol = "grpc";
  d5.source_app = "app_source";
  d6.request_protocol = "grpc";
  d6.source_app = "app_source";
  d6.source_version = "v2";
  d7.outbound = true;d7.request_protocol = "grpc";
  d7.source_app = "app_source";d7.source_version = "v2";
  d8.outbound = true;d8.request_protocol = "grpc";
  d8.source_app = "app_source";d8.source_version = "v2";
/=============================/
Thx.